### PR TITLE
added postgres datbase using chart and aso

### PR DESCRIPTION
**Pull request by cpareek**
**Before creating a pull request make sure that:**

- [ X ] commit messages are meaningful and follow good commit message guidelines

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-14838

### Change description ###

Added postgres databases using chart-postgresql

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: charts/sscs-tribunals-api/values.ccd.preview.template.yaml [hunk 1]
```diff
@@ -768,3 +768,20 @@ servicebus:
         subscriptionNeeded: yes
     queues:
       - name: sscs-orchestrator-queue-preview
+
+#Add postgres sql databases
+postgresql:
+  enabled: true
+  flexibleserver: sscs-flexibleserver-postgres-preview
+  #Each topic/queue matches up to one used in the config above
```
Comment (timja) [inline charts/sscs-tribunals-api/values.ccd.preview.template.yaml:776]:
```suggestion
```

File: charts/sscs-tribunals-api/values.ccd.preview.template.yaml [hunk 2]
```diff
@@ -768,3 +768,20 @@ servicebus:
         subscriptionNeeded: yes
     queues:
       - name: sscs-orchestrator-queue-preview
+
+#Add postgres sql databases
```
Comment (timja) [inline charts/sscs-tribunals-api/values.ccd.preview.template.yaml:772]:
```suggestion
```

File: charts/sscs-tribunals-api/values.ccd.preview.template.yaml [hunk 3]
```diff
@@ -146,20 +146,20 @@ ccd:
...
+  #         init.sql: |
+  #           CREATE DATABASE "data-store" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
+  #           CREATE DATABASE "definition-store" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
+  #           CREATE DATABASE "user-profile" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
+  #           CREATE DATABASE "draftstore" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
+  #           CREATE DATABASE "evidence" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
+  #           CREATE DATABASE "annotation" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
+  #           CREATE DATABASE "role_assignment" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
+  #           CREATE DATABASE "hmc-cft-hearing-service" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
+  #           CREATE DATABASE "sscsjobscheduler" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
```
Comment (timja) [inline charts/sscs-tribunals-api/values.ccd.preview.template.yaml:162]: this should be removed.

you will then need to reconfigure the app(s) to use the created secret(s).

see an example in here:
https://github.com/hmcts/cnp-plum-recipes-service/pull/628/files#diff-b4c087d8f34d8f675dd49475fcb55e17dbcad4ff3a0f266ae7f048810b2d4845

**Review by timja — CHANGES_REQUESTED** [review]

File: charts/sscs-tribunals-api/values.ccd.preview.template.yaml (hunk 3 above)
Comment (cpareek) [reply charts/sscs-tribunals-api/values.ccd.preview.template.yaml:162]: so the postgres server password has been set here using secrets - 
https://github.com/hmcts/cnp-flux-config/blob/83b0c90a67f69e75cb7b77cf2be3c48d5441982d/apps/sscs/preview/aso/sscs-postgres.yaml#L18

Comment (cpareek) [reply charts/sscs-tribunals-api/values.ccd.preview.template.yaml:162]: I will remove above commented lines, sure

File: charts/sscs-tribunals-api/values.ccd.preview.template.yaml [hunk 4]
```diff
@@ -139,27 +139,11 @@ ccd:
...
       enabled: false
     elastic:
       enabled: true
 
-  postgresql:
-    primary:
-      initdb:
-        scripts:
-          init.sql: |
-            CREATE DATABASE "data-store" WITH OWNER = hmcts ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
```
Comment (timja) [inline charts/sscs-tribunals-api/values.ccd.preview.template.yaml:154]: all the apps still need to be configured to use the secret you've created to auth to the dbs and to use the right db url etc

Comment (cpareek) [reply charts/sscs-tribunals-api/values.ccd.preview.template.yaml:154]: sure - thanks -checking that now

File: charts/sscs-tribunals-api/values.ccd.preview.template.yaml [hunk 5]
```diff
@@ -444,7 +419,7 @@ ccd:
       memoryRequests: '1524Mi'
       cpuRequests: '900m'
       environment:
-        DATA_STORE_DB_HOST: ${SERVICE_NAME}-postgresql
+        DATA_STORE_DB_HOST: "sscs-flexibleserver-postgres-preview.postgres.database.azure.com"
```
Comment (timja) [inline charts/sscs-tribunals-api/values.ccd.preview.template.yaml:422]: you should be setting the global value instead in the global section:
https://github.com/hmcts/sscs-tribunals-case-api/blob/26507b9fd183dc869ae2aa58056ee7daa85b097b/charts/sscs-tribunals-api/values.ccd.preview.template.yaml#L148

removing any references to db host that belong to the ccd chart as that will use the global and in any SSCS dbs you should change to using the global url so it's only referenced once (may need to use tpl actually i might have been wrong)

**Review by timja — CHANGES_REQUESTED** [review]

File: charts/sscs-tribunals-api/values.ccd.preview.template.yaml (hunk 5 above)
Comment (cpareek) [reply charts/sscs-tribunals-api/values.ccd.preview.template.yaml]: done

**Review by timja — CHANGES_REQUESTED** [review]
PR needs to be labelled with `pr-values:ccd` and a new build triggered

https://github.com/hmcts/sscs-tribunals-case-api/blob/ca690eff64ae7613e05fed28a116822b10ec5971/charts/sscs-tribunals-api/values.ccd.preview.template.yaml#L1C1-L1C1

**Comment by cpareek** [comment]
> what are you after here in a review? changes look reasonable but the build is failing

Yes - trying to fix the build, something to do around the dependancy so trying to get that fix.

**Pull request closed without merging**